public/app/features/correlations/components/Wizard/wizardContext.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { createContext, PropsWithChildren, useContext, useState } from 'react';
+import { FieldValues } from 'react-hook-form';
+
+export type WizardContextProps<T> = {
+  currentPage: number;
+  nextPage: () => void;
+  prevPage: () => void;
+  isLastPage: boolean;
+  onSubmit: (data: T) => void;
+  CurrentPageComponent: React.ComponentType;
+};
+
+export const WizardContext = createContext<WizardContextProps<FieldValues>>({
+  currentPage: 0,
+  nextPage: () => {},
+  prevPage: () => {},
+  isLastPage: true,
+  onSubmit: () => {},
+  CurrentPageComponent: () => null,
+});
+
+/**
+ * Dependencies provided to Wizard component required to build WizardContext
+ */
+type WizardContextProviderDeps<T> = {
+  pages: React.ComponentType[];
+  onSubmit: (data: T) => void;
+};
+
+/**
+ * Context providing current state and logic of a Wizard. Can be used by pages and navigation components.
+ */
+export function WizardContextProvider<T>(props: PropsWithChildren<WizardContextProviderDeps<T>>) {
+  const [currentPage, setCurrentPage] = useState(0);
+  const { pages, onSubmit, children } = props;
+
+  const context: WizardContextProps<T> = {
+    currentPage,
+    CurrentPageComponent: pages[currentPage],
+    isLastPage: currentPage === pages.length - 1,
+    nextPage: () => setCurrentPage(currentPage + 1),
+    prevPage: () => setCurrentPage(currentPage - 1),
+    onSubmit,
+  };
+
+  // @ts-ignore
+  return <WizardContext.Provider value={context}>{children}</WizardContext.Provider>;
+}
+
+export const useWizardContext = () => {
+  return useContext(WizardContext);
+};
```

Comment:
i usually like to do something like 

```
export const WizardContext = createContext<WizardContextProps<FieldValues> | undefined>(undefined);

// ...

export const useWizardContext = () => {
    const ctx = useContext(WizardContext);
    
    if(!ctx){
      throw new Error("useWizardContext must be used within a WizardContextProvider")
    }
    return ctx;
};
```

so that you don't have to provide a default value that is meaningless and at the same time you can prevent consumers from using this when it'd have no effect. wdyt?